### PR TITLE
[Tests-Only] Add expire share tests

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
- added acceptance tests for different shares describing how they behave after getting expired

## Related Issue

- https://github.com/owncloud/core/issues/36550

## How Has This Been Tested?

- manually
- ci

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiFederationToRoot2/federated.feature [hunk 1]
```diff
@@ -668,3 +668,20 @@ Feature: federated
       | ocs_api_version | http_status_code |
       | 1               | 200              |
       | 2               | 404              |
+
+
+  Scenario: set a federated user share to expire yesterday and verify that it is not accessible
```
Comment (dpakach) [inline tests/acceptance/features/apiFederationToRoot2/federated.feature:673]: I don't think we are testing setting the expiration date, but rather accessing the share.
```suggestion
  Scenario: Accessing the share that is expired should not be possible
```
Also same for other scenarios

File: tests/acceptance/features/apiFederationToRoot2/federated.feature [hunk 2]
```diff
@@ -668,3 +668,20 @@ Feature: federated
...
+    Given using OCS API version "2"
+    And using server "REMOTE"
+    And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "/textfile0.txt"
+    And user "Alice" from server "REMOTE" has shared "/textfile0.txt" with user "Brian" from server "LOCAL" with expiry "+5 days"
```
Comment (dpakach) [inline tests/acceptance/features/apiFederationToRoot2/federated.feature:677]:
```suggestion
    And user "Alice" from server "REMOTE" has shared "/textfile0.txt" with user "Brian" from server "LOCAL" with expiry date of "+5 days"
```

File: tests/acceptance/features/apiShareCreateSpecialToRoot1/createShareExpirationDate.feature [hunk 3]
```diff
@@ -764,3 +764,61 @@ Feature: a default expiration date can be specified for shares with users or gro
...
+    And user "Brian" has been added to group "brand-new-group"
+    And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "/textfile0.txt"
+    And user "Alice" has created a public link share with settings
+      | path        | /textfile0.txt  |
+      | shareType   | group           |
+      | shareWith   | brand-new-group |
+      | expireDate  | +15 days        |
+    And the administrator has expired the last created share using the testing API
+    When the public accesses the preview of file "textfile0.txt" from the last shared public link using the sharing API
+    And the HTTP status code should be "404"
```
Comment (Talank) [inline tests/acceptance/features/apiShareCreateSpecialToRoot1/createShareExpirationDate.feature:822]:
```suggestion
    Then the HTTP status code should be "404"
```

File: tests/acceptance/features/bootstrap/AppConfigurationContext.php [hunk 4]
```diff
@@ -506,6 +506,49 @@ public function theTrustedServerListShouldBeEmpty() {
...
+
+	/**
+	 * @When the administrator expires the last created share using the testing API
+	 *
+	 * @return void
+	 */
+	public function theAdministratorExpiresTheLastCreatedShare() {
+		$this->expireLastCreatedUserShare();
+	}
+
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/AppConfigurationContext.php:551]: these should be removed from testing app context

File: tests/acceptance/features/apiFederationToRoot2/federated.feature [hunk 5]
```diff
@@ -668,3 +668,20 @@ Feature: federated
       | ocs_api_version | http_status_code |
       | 1               | 200              |
       | 2               | 404              |
+
+
```
Comment (phil-davis) [inline tests/acceptance/features/apiFederationToRoot2/federated.feature:672]:
```suggestion
  @skipOnOcV10.6 @skipOnOcV10.7
```

File: tests/acceptance/features/apiShareCreateSpecialToRoot1/createShareExpirationDate.feature [hunk 6]
```diff
@@ -764,3 +764,60 @@ Feature: a default expiration date can be specified for shares with users or gro
       | ocs_api_version |
       | 1               |
       | 2               |
+
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareCreateSpecialToRoot1/createShareExpirationDate.feature:767]:
```suggestion

  @skipOnOcV10.6 @skipOnOcV10.7
```

File: tests/acceptance/features/apiShareCreateSpecialToRoot1/createShareExpirationDate.feature [hunk 7]
```diff
@@ -764,3 +764,60 @@ Feature: a default expiration date can be specified for shares with users or gro
...
+      | expireDate  | +15 days       |
+    And the administrator has expired the last created share using the testing API
+    When user "Alice" gets the info of the last share using the sharing API
+    Then the OCS status code should be "404"
+    And the HTTP status code should be "200"
+    And user "Alice" should not see the share id of the last share
+    And user "Brian" should not see the share id of the last share
+    And as "Brian" file "/textfile0.txt" should not exist
+    And as "Alice" file "/textfile0.txt" should exist
+
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareCreateSpecialToRoot1/createShareExpirationDate.feature:786]:
```suggestion

  @skipOnOcV10.6 @skipOnOcV10.7
```

File: tests/acceptance/features/apiShareCreateSpecialToRoot1/createShareExpirationDate.feature [hunk 8]
```diff
@@ -764,3 +764,60 @@ Feature: a default expiration date can be specified for shares with users or gro
...
+      | expireDate  | +15 days        |
+    And the administrator has expired the last created share using the testing API
+    When user "Alice" gets the info of the last share using the sharing API
+    Then the OCS status code should be "404"
+    And the HTTP status code should be "200"
+    And user "Alice" should not see the share id of the last share
+    And user "Brian" should not see the share id of the last share
+    And as "Brian" file "/textfile0.txt" should not exist
+    And as "Alice" file "/textfile0.txt" should exist
+
```
Comment (phil-davis) [inline tests/acceptance/features/apiShareCreateSpecialToRoot1/createShareExpirationDate.feature:807]:
```suggestion

  @skipOnOcV10.6 @skipOnOcV10.7
```

**Review by phil-davis — COMMENTED** [review]
We need to skip all these new scenarios on old oC10, because the new core logic to set an expiration date in the past does not exist there.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/30507/108/18
```
runsh: Total unexpected failed scenarios throughout the test run:
apiFederationToRoot2/federated.feature:673
```

The `skipOnOcV10.7` tag did not work. That scenario should have been skipped. I think the logic for skipping might not be good enough, because in this case the "local" server is core master 10.8 and the "federated" server is 10.7.

@kiranparajuli589 please have a look - maybe we just need to add some `skipOnFedOcV10.7` and ``skipOnFedOcV10.6` tags to the scenario(s) that use a federated server? (Those `skipOnFed*` tags are mentioned in `run.sh`)

File: tests/acceptance/features/bootstrap/AppConfigurationContext.php (hunk 4 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/AppConfigurationContext.php:551]: in pr https://github.com/owncloud/testing/pull/174

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/30508/109/18
That pipeline is running against a 10.5.0 federated server. We can update that to a 10.6.0 federated server - I did that in the last commit.

**Pull request merged**